hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -206,6 +208,14 @@ def decorator(strategy_definition):
 
         @proxies(strategy_definition)
         def accept(*args, **kwargs):
+            if not force_lazy:
+                # Why not try this unconditionally?  Because we'd end up with very
+                # deep nesting of recursive strategies - better to be lazy unless we
+                # *know* that eager evaluation is the right choice.
+                try:
+                    return strategy_definition(*args, **kwargs)
+                except Exception:
```

Comment:
Might be worth a comment saying why we have this here? Silently swallowing the error is a bit of a red flag normally (I know it will be reraised later due to laziness but it'd be nice to say that)